src/functionModules/metamask/MetamaskModule.ts

```diff
@@ -41,29 +49,54 @@ export default class MetamaskModule {
     return enabled
   }
 
-  public static async send(web3: Web3,
-                           opts: Transaction,
-                           flashesModule: FlashesModule,
-                           newListingModule: NewListingModule,
-                           uploadModule: UploadModule,
-                           successCallback: (response: any,
-                                             flashesModule: FlashesModule,
-                                             newListingModule: NewListingModule,
-                                             uploadModule: UploadModule) => void) {
-
-    opts.gas = web3.utils.toHex(opts.gas)
-    opts.gasPrice = web3.utils.toHex(opts.gasPrice)
-
-    ethereum.sendAsync({
-      method: 'eth_sendTransaction',
-      params: [opts], // NOTE: do not miss that this is an array of 1
-      from: ethereum.selectedAddress,
+  public static async buildAndSendTransaction(
+    account: string,
+    method: [Transaction, TransactOpts],
+    contractAddress: string,
+    appStore: Store<any>,
+    success: (
+      response: any,
+      appStore: Store<any>) => void) {
+
+    const web3Module = getModule(Web3Module, appStore)
+
+    //  get gas estimate using method[0]
+    // @ts-ignore
+    const estmatedGas = await method[0].estimateGas({from: account})
+    const unsigned = await buildTransaction(web3Module.web3, method)
+    unsigned.to = contractAddress
+    unsigned.value = ZERO_HASHED
+    // MM ignores any nonce, let's just remove it
+    delete unsigned.nonce
+    // take the larger of the two gas estimates to be safe
+    unsigned.gas = Math.max(estmatedGas, unsigned.gas)
+
+    MetamaskModule.send(web3Module.web3, unsigned, appStore, success)
+  }
+
+  public static async send(
+    web3: Web3,
+    unsignedTransaction: Transaction,
+    appStore: Store<any>,
+    successCallback: (
+      response: any,
+      appStore: Store<any>) => void) {
+
+    unsignedTransaction.gas = web3.utils.toHex(unsignedTransaction.gas)
+    unsignedTransaction.gasPrice = web3.utils.toHex(unsignedTransaction.gasPrice)
+
+    ethereum.sendAsync(
+      {
+        method: 'eth_sendTransaction',
+        params: [unsignedTransaction], // NOTE: do not miss that this is an array of 1
+        from: ethereum.selectedAddress,
       },
       (err: any, res: any) => {
         if (err) {
+          const flashesModule  = getModule(FlashesModule, appStore)
           flashesModule.append(new Flash(err, FlashType.error))
         } else if (successCallback) {
```

Comment:
can just be `else`

Reply:
this block is still tentative (for me the giveaway is the `err` and `res` param names).  let's leave it for now as the pattern is still in flux and is likely to change when the error handling gets implemented.